readthedocs/restapi/client.py

```diff
@@ -37,12 +37,14 @@ def dumps(self, data):
 
 def setup_api():
     session = requests.Session()
-    session.mount(
-        API_HOST,
-        host_header_ssl.HostHeaderSSLAdapter(
-            max_retries=3,
-        ),
-    )
+    if API_HOST.startswith('https'):
+        # Only use the HostHeaderSSLAdapter for HTTPS connections
+        session.mount(
+            API_HOST,
+            host_header_ssl.HostHeaderSSLAdapter(
+                max_retries=3,
+            ),
+        )
```

Comment:
`max_retries` was from the original adapter, we should add this with the standard http adapter if the connection is not ssl.

Reply:
That was not there prior to the [PR that added the `HostHeaderSSLAdapter`](https://github.com/rtfd/readthedocs.org/pull/4423/files#diff-954ebd960ddf2636d3a2dbbc21d39e73R40). Do you think it is necessary to support retries in dev?